src/components/ProductList.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import fetchProducts from "utils/api";
+import { getBreakpoint, updateProductsPerPage } from "utils/checkDevice";
+import Product from "./Product";
+import DropDown from "./DropDown";
+import Search from "./Search";
+import PrimaryButton from "./PrimaryButton";
+import Paginations from "./Paginations";
+
+function ProductList() {
+  const navigate = useNavigate();
+  const initialDevice = getBreakpoint(window.innerWidth);
+  const initialperPage = updateProductsPerPage(initialDevice);
+  const [products, setProducts] = useState([]); // 서버에서 받아올 Proudcts를 할당할 state
+  const [productsPerPage, setProductsPerPage] = useState(initialperPage); // 반응형에 따라 보여줄 Product 수를 할당할 state
+  const [device, setDevice] = useState(initialDevice); // 반응형 ('pc' | 'tablet' | 'mobile')
+  const [order, setOrder] = useState("recent"); // 데이터 정렬을 위한 queryParam [orderBy]
+  const [totalPage, setTotalPage] = useState(0); // Pagination을 위한 전체 페이지 수를 할당할 state
```

Comment:
이 값도 products 라는 state 만 있어도 충분히 계산 가능한 값 같습니다. products 라는 state가 변경되면 해당 컴포넌트가 리렌더링 되면서 내부 값들도 새롭게 연산할것이기 때문에 totalPage는 state로 관리할 필요 없이 일반적인 변수로 할당해도 좋을 것 같아요.
```suggestion
  const productsCountperPage = updateProductsPerPage(currentDevice);
  const totalPage = Math.ceil(products.length / productsCountperPage );
```